### DISCUSSION TITLE
How does Component detect changes to children's properties without using MutationObserver?

### DISCUSSION THREAD

**Discussion by GakkiiSmile** [solidjs/solid#1644] [General]
I think SolidJS should provide an API similar to MutationObserver. In one scenario, my button component is intended for use by others, and it proactively optimizes the display of button text, so it needs to detect changes to its child elements.
 [example](https://playground.solidjs.com/anonymous/2e68eb5c-b1f7-4376-ab87-ba5284239b58)

[https://github.com/ryansolid/dom-expressions/blob/main/packages/dom-expressions/src/client.js#L89](https://github.com/ryansolid/dom-expressions/blob/main/packages/dom-expressions/src/client.js#L89)
like this 
```typescript
export function setAttribute(node, name, value) {
 if (value == null) {
  node.removeAttribute(name);
  node.__INTERNAL_NOTIFY?.({
    type: "removeAttribute",
    data: {},
  });
} else {
  node.setAttribute(name, value);

  node.__INTERNAL_NOTIFY?.({
    type: "setAttribute",
    data: {},
  });
}
```

**Comment by atk** [comment]
There is https://github.com/solidjs-community/solid-primitives/tree/main/packages/mutation-observer#readme to fulfil your need.
